plugin/rewrite/name.go

```diff
@@ -194,3 +196,16 @@ func (rule *substringNameRule) GetResponseRule() ResponseRule { return ResponseR
 
 // GetResponseRule return a rule to rewrite the response with.
 func (rule *regexNameRule) GetResponseRule() ResponseRule { return rule.ResponseRule }
+
+// validName returns true if s is valid domain name and shortern than 256 characters.
+func validName(s string) bool {
+	_, ok := dns.IsDomainName(s)
+	if !ok {
+		return false
+	}
+	if len(dns.Name(s).String()) > 255 {
+		return false
+	}
+
```

Comment:
if you return an error with the exact pb, it will be more easy to find out the issue and fix it.
=> err("name is not a valid domain name")
=> err("name is too long (over 255 chars)")

Reply:
> if you return an error with the exact pb, it will be more easy to find out the issue and fix it.

@fturib , good point 👍